### PR TITLE
chore: tweak mergify configuration

### PR DESCRIPTION
**Pull request by RomainMuller**
- Require the TravisCI has passed before auto-merging a PR. This is
  required to ensure the `jsii/superchain` Docker image was correctly
  built and tested before going forward.
- Drop stale "approved" reviews, but keep stale "changes requested"
  reviews, so that people who requested changes can come and verify that
  the changes were indeed applied.
- Remove the "Pull Request Checklist" message as it was emitted multiple
  times and that god somewhat irritating. Didn't seem to find a clean
  way to prevent multiple emission here.
- Tweaked message for when a PR is ready for automatic merging.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -4,46 +4,46 @@ pull_request_rules:
...
       - -merged
       - -closed
-      - "#approved-reviews-by>=1"
       - -approved-reviews-by~=author
+      - "#approved-reviews-by>=1"
+      - "#review-requested=0"
       - "#changes-requested-reviews-by=0"
       - "#commented-reviews-by=0"
       - status-success=AWS CodeBuild us-east-1 (jsii-build)
+      - status-success=continuous-integration/travis-ci/pr
```
Comment (eladb) [inline .mergify.yml]: Why is that needed?

Comment (RomainMuller) [reply .mergify.yml]: Currently this is what's checking the `jsii/superchain` Docker image builds & can build `jsii` itself. It'll become redundant once the handling of the Docker image is moved out to the `CodePipeline` setup, but I haven't got around to cater for that just yet.
